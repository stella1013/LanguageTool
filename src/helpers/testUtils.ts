//import { ConnectedRouter } from 'connected-react-router';
//import React from 'react';
//import { Provider } from 'react-redux';
////import { mergeDeepRight } from 'ramda';
//import rootReducer from '../redux/reducers';
////import  createStoreWithMiddleWare  from '../redux/store';
//import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
//import { AppState } from '../redux/store';
//import ReduxThunk  from 'redux-thunk';
//import { createStore, applyMiddleware } from 'redux';
//import rootReducer from '../redux/reducers';
//import thunk from 'redux-thunk';

/*
	* Returns Shallow Wrapper containing node(s) with the given data-test value
	* @param {Shallow Wrapper} - Enzyme shallow wrapper to search within
	* @param {string} val - Value of data-test attribute for search.
	* @returns {Shallow Wrapper}
	*/ 
	export const findByTestAttr = (wrapper:any, val:string) => {
		return wrapper.find(`[data-test="${val}"]`);
	}

	export const storeFactory = (initialState:any) =>{
		const createMockStore = configureMockStore();
		return createMockStore(initialState);
	}
	/*
	export const makeMountRender = (Component:any, defaultProps = {}) => {
		return (customProps = {}) => {
		  const props = {
			...defaultProps,
			...customProps
		  };
		  return mount(<Component {...props} />);
		};
	  };

	  export const makeStore = (customState = {}) => {
		const root = rootReducer({}, { type: '@@INIT' });
		const state = mergeDeepRight(root, customState);
		return createStoreWithMiddleWare(rootReducer, state);
	  };
	  export const snapshotify = (reactWrapper:any) => {
		return reactWrapper.html();
	  };

	  export const reduxify = (Component, props = {}, state = {}) => {
		return function reduxWrap() {
		  return (
			<Provider store={makeStore(state)}>
			  <ConnectedRouter history={history}>
				<Component {...props} />
			  </ConnectedRouter>
			</Provider>
		  );
		}
	  };

*/