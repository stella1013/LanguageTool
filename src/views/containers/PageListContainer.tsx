import * as React from 'react';
import { connect } from 'react-redux';
import { AppState } from '../../redux/store';
import Form from './FormContainer';
import PageComponent from '../components/PageComponent';
import PageNavigationContainer from './PageNavigationContainer';


interface PageState{
	pages:[]
}
class PageListContainer extends React.Component<any, PageState> {
	constructor(props:any){
		super(props);
		this.state = {
			pages:[]
		}
	}
	componentDidMount(){
		this.setState({pages:this.props.data});
	}
	
	render() {
		
		let pages: any = null;
		if (this.state.pages.length > 0) {
			
			pages = (
				<div>
					{this.state.pages.map((singlePage: any, index: any) => {
						return (

								index === this.props.pagecontrol ? <PageComponent key={index} id={index+1} data-test="component-page" >
								<Form
									formScaffold={singlePage.page}
									//changed={this.props.changed}
								/>
							</PageComponent>
							: null
						);
					})}
				</div>
			);
		}
		return (
			<div className="PageListContainer" data-test="component-pagelist">
				{pages}
				<PageNavigationContainer/>
			</div>
			
		);
	}
}

const mapStateToProps = (state: AppState) => {
	return {
		data: state.data.pages,
		pagecontrol: state.pagelist.currentPage
	};
};
const mapDispatchToProps = (dispatch: any) => {
	return {
		//showModal: (showModal:boolean) => dispatch(modalOperations.showModal(showModal))
	};
};
type ReduxType = ReturnType<typeof mapStateToProps> &
	ReturnType<typeof mapDispatchToProps>;
export default connect(
	mapStateToProps,
	mapDispatchToProps
)(PageListContainer);
