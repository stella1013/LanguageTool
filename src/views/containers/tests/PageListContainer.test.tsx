import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import configureMockStore from 'redux-mock-store';
import { findByTestAttr } from '../../../helpers/testUtils';
import thunk from 'redux-thunk';
import PageListContainer from '../PageListContainer';
import { Provider } from 'react-redux';

describe('Page List', () => {
	/*
	 * Factory function to create a Shallow Wrapper for the App Component
	 * @function setup
	 * @param {object} props - component props
	 * @param {object} state - initial state
	 * @returns {Shallow Wrapper}
	 */
	const createMockStore = configureMockStore([thunk]);
	const initialState = {};
	
	let wrapper: any;
	describe('rendering', () => {
		let props = {
			appTitle: 'title'
		};
		beforeEach(() => {
			const store = createMockStore(initialState);
			wrapper = shallow(
				<Provider store={store}>
					<PageListContainer />
				</Provider>
			);
		});
		it('renders without crashing', () => {
			const component = findByTestAttr(wrapper, 'component-pagelist');
			expect(component).toBeTruthy();
		});

		it('render matches snapshot', () => {
			expect(wrapper).toMatchSnapshot();
		});
	
	});

	describe('state updates', () => {
		describe('when navigating pages', () => {
			/*
			it('next button it navigates to the next page', () => {
				const currentPage = 1;
				const component = findByTestAttr(wrapper, 'component-pagenav');
				expect(currentPage).toBe(2);
			});
			it('back button it navigates to the previous page', () => {
				const currentPage = 2;
				const component = findByTestAttr(wrapper, 'component-pagenav');
				expect(currentPage).toBe(1);
			});
			it('initial `state` of current page at 1', () => {
				expect(wrapper.state('currentPage')).toEqual(1);
			});
			it('renders navigation component', () => {
				const component = findByTestAttr(wrapper, 'component-pagenav');
				expect(component.length).toBe(1);
			});
			*/
		});
	});
});
