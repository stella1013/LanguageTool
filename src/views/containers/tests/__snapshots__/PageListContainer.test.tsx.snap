// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page List rendering render matches snapshot 1`] = `ShallowWrapper {}`;
